security/pkg/k8s/chiron/utils.go

```diff
@@ -253,9 +253,12 @@ func readSignedCsr(client clientset.Interface, csr string, watchTimeout time.Dur
 		FieldSelector: selector,
 	})
 	if l != nil && len(l.Items) > 0 {
-		reqSigned := l.Items[0]
-		if reqSigned.Status.Certificate != nil {
-			return reqSigned.Status.Certificate, nil
+		for _, reqSigned := range l.Items {
+			// technically, our fieldSelector means there should only be one item,
```

Comment:
I don't see where this happens

I added

```
		if len(l.Items) > 1 {
			panic(fmt.Sprintf("%+v", l.Items))
		}
```

to the code (both in this PR and on master) and it never fails?

Reply:
That is correct.  This test used to create 2 CSRs, but now only creates 1.  I don't think it is safe to depend on list() only ever returning 1, as some tests may create multiple CSRs, not to mention in production if api server were to return more than one, a critical piece of security infrastructure would misbehave.

Reply:
I think it is safe to assume in prod tbh, the entire k8s ecosystem would break if that invariant were broken (including kubelet, which this code is mostly copy+pasted from).

Reply:
BTW, why didnot we use `CertificatesV1().CertificateSigningRequests().Get(` previously

BTW, k8s provides a `func WaitForCertificate(ctx context.Context, client clientset.Interface, reqName string, reqUID types.UID) (certData []byte, err error) {
`, so we donot need to copy this function at all